sdk/data/azcosmos/cosmos_location_cache.go

```diff
@@ -50,20 +50,20 @@ type locationCache struct {
 	locationInfo                      databaseAccountLocationsInfo
 	defaultEndpoint                   url.URL
 	enableEndpointDiscovery           bool
-	useMultipleWriteLocations         bool
 	locationUnavailabilityInfoMap     map[url.URL]locationUnavailabilityInfo
 	mapMutex                          sync.RWMutex
 	lastUpdateTime                    time.Time
 	enableMultipleWriteLocations      bool
 	unavailableLocationExpirationTime time.Duration
 }
 
-func newLocationCache(prefLocations []string, defaultEndpoint url.URL) *locationCache {
+func newLocationCache(prefLocations []string, defaultEndpoint url.URL, enableEndpointDiscovery bool) *locationCache {
 	return &locationCache{
 		defaultEndpoint:                   defaultEndpoint,
 		locationInfo:                      *newDatabaseAccountLocationsInfo(prefLocations, defaultEndpoint),
 		locationUnavailabilityInfoMap:     make(map[url.URL]locationUnavailabilityInfo),
 		unavailableLocationExpirationTime: defaultExpirationTime,
+		enableEndpointDiscovery:           enableEndpointDiscovery,
```

Comment:
Why is this flag here if this is something used in the client retry policy? Can we not pass it as the policy constructor?

Reply:
I was just aligning to other SDKs:
Java:
<img width="212" alt="image" src="https://github.com/Azure/azure-sdk-for-go/assets/30335873/55bea913-09b0-4bc8-844d-7fafd06eedf1">

Python:
<img width="232" alt="image" src="https://github.com/Azure/azure-sdk-for-go/assets/30335873/d801bcfb-9266-4152-9f2e-0b74159a4095">


will make it a policy-only thing though

Reply:
If we are using it in the location cache for something, then leave it. I'm just not sure if we area